ironic-python-agent/0001-Bumping-version-of-debian-to-stretch-adding-user-cor.patch

```diff
@@ -0,0 +1,124 @@
+ 4 files changed, 37 insertions(+), 7 deletions(-)
+ create mode 100755 get_megacli.sh
+
+diff --git a/Dockerfile b/Dockerfile
+index 8c7a56a..d8f02b4 100644
+--- a/Dockerfile
++++ b/Dockerfile
+@@ -1,4 +1,4 @@
+-FROM debian:jessie
++FROM debian:stretch
+ 
+ # The add is before the RUN to ensure we get the latest version of packages
+ # Docker will cache RUN commands, but because the SHA1 of the dir will be
+@@ -13,20 +13,25 @@ ADD . /tmp/ironic-python-agent
+ # 1.6. Using the ARG command will be a much cleaner solution.
+ COPY proxy.sh /usr/bin/proxy.sh
+ 
++ADD get_megacli.sh /tmp/get_megacli.sh
++
+ # Ensure we hit a single mirror for builds, since httpredir is flakey
+ RUN sed -i 's/httpredir/http.us/g' /etc/apt/sources.list
+ 
+ # Add 'backports' for qemu-utils
+-RUN echo 'deb http://http.us.debian.org/debian jessie-backports main' > /etc/apt/sources.list.d/backports.list
++RUN echo 'deb http://http.us.debian.org/debian stretch-backports main' > /etc/apt/sources.list.d/backports.list
++
++ENV DEBIAN_FRONTEND noninteractive
+ 
+ # Install requirements: Python for ironic-python-agent, others for putting an
+ # image on disk
+ RUN proxy.sh apt-get update && \
+     proxy.sh apt-get -y upgrade && \
+-    proxy.sh apt-get install -y --no-install-recommends gdisk python2.7 python2.7-dev \
+-        python-pip qemu-utils parted hdparm util-linux genisoimage git gcc \
+-        bash coreutils tgt dmidecode ipmitool psmisc dosfstools && \
+-    proxy.sh apt-get --only-upgrade -t jessie-backports install -y qemu-utils
++    proxy.sh apt-get install -y --no-install-recommends gdisk python2.7 python2.7-dev python2.7-setuptools lvm2 \
++        python-pip qemu-utils parted hdparm util-linux genisoimage git gcc udev mdadm wget unzip alien libguestfs-tools \
++        bash coreutils tgt dmidecode ipmitool psmisc dosfstools pciutils && \
++    proxy.sh apt-get install -y linux-image-amd64 && \
++    proxy.sh apt-get --only-upgrade -t stretch-backports install -y qemu-utils
+ 
+ # Some cleanup
+ RUN proxy.sh apt-get -y autoremove && \
+@@ -44,6 +49,9 @@ RUN proxy.sh pip install -c /tmp/ironic-python-agent/upper-constraints.txt --no-
+ # This will succeed because all the dependencies were installed previously
+ RUN proxy.sh pip install -c /tmp/ironic-python-agent/upper-constraints.txt --no-cache-dir /tmp/ironic-python-agent
+ 
++# Install megacli
++RUN /tmp/get_megacli.sh
++
+ # Remove no longer needed packages
+ # NOTE(jroll) leave git to avoid strange apt issues in downstream Dockerfiles
+ # that may inherit from this one.
+diff --git a/get_megacli.sh b/get_megacli.sh
+new file mode 100755
+index 0000000..d296914
+--- /dev/null
++++ b/get_megacli.sh
+@@ -0,0 +1,16 @@
++#!/usr/bin/env sh
++
++VERSION=${VERSION:-'8-07-14'}
++# replace - with .
++RPM_VERSION=${VERSION//\-/\.}
++# get MegaCli
++wget https://docs.broadcom.com/docs-and-downloads/raid-controllers/raid-controllers-common-files/${VERSION}_MegaCLI.zip
```

Comment:
Wondering if we have an internal location we'll want to get this from. Not a big deal right now as we'll also need to figure out how to use our internal pypi mirrors and such with this builder.

Reply:
Summoning @epim for this one, though I don't think it's a blocker.